asana/asana.go

```diff
@@ -189,8 +183,39 @@ type (
 
 	// Errors always has at least 1 element when returned.
 	Errors []Error
+
+	RequestError struct {
+		Err      error
+		Response *http.Response
```

Comment:
I'm actually wondering if it might be better to embed the body and maybe headers of the response rather than the entire `Response` object...

Reply:
yeah, I think exposing the entire response object is probably a bad thing. Ideally consumers of the SDK don't need to know about the http layer.

Modeling it off the [GoogleAPI](https://godoc.org/google.golang.org/api/googleapi#Error) error might be nice. They have an abstraction which picks out the error code and a message, but allows access to the full body. I wouldn't include the `http.Header` unless there is very general error information stored there. Nicer to unpack into a strongly typed field.